crates/store/re_sorbet/src/chunk_batch.rs

```diff
@@ -51,58 +219,109 @@ impl From<ChunkBatch> for ArrowRecordBatch {
-                        "ChunkBatch version mismatch. Expected {:?}, got {batch_version:?}",
-                        Self::CHUNK_METADATA_VERSION
-                    );
-                }
-            }
-            metadata.insert(
-                Self::CHUNK_METADATA_KEY_VERSION.to_owned(),
-                Self::CHUNK_METADATA_VERSION.to_owned(),
-            );
+    /// Will automatically wrap data columns in `ListArrays` if they are not already.
+    fn try_from(batch: &ArrowRecordBatch) -> Result<Self, Self::Error> {
+        re_tracing::profile_function!();
+
+        let batch = make_all_data_columns_list_arrays(batch);
+
+        let chunk_schema = ChunkSchema::try_from(batch.schema_ref().as_ref())?;
+
+        for (field, column) in itertools::izip!(chunk_schema.arrow_fields(), batch.columns()) {
+            debug_assert_eq!(field.data_type(), column.data_type());
         }
 
-        let entity_path =
-            if let Some(entity_path) = metadata.get(Self::CHUNK_METADATA_KEY_ENTITY_PATH) {
-                EntityPath::parse_forgiving(entity_path)
-            } else {
-                return Err(ChunkBatchError::MissingRecordBatchMetadataKey(
-                    Self::CHUNK_METADATA_KEY_ENTITY_PATH,
-                ));
-            };
+        // Extend with any metadata that might have been missing:
+        let mut arrow_schema = ArrowSchema::clone(batch.schema_ref().as_ref());
+        arrow_schema
+            .metadata
+            .extend(chunk_schema.arrow_batch_metadata());
 
-        Ok(Self { entity_path, batch })
+        let batch = ArrowRecordBatch::try_new_with_options(
+            arrow_schema.into(),
+            batch.columns().to_vec(),
+            &RecordBatchOptions::default().with_row_count(Some(batch.num_rows())),
+        )
+        .expect("Can't fail");
+
+        Ok(Self {
+            schema: chunk_schema,
+            batch,
+        })
     }
 }
 
-/// ## Metadata keys for the record batch metadata
-impl ChunkBatch {
-    /// The key used to identify the version of the Rerun schema.
-    const CHUNK_METADATA_KEY_VERSION: &'static str = "rerun.version";
+/// Make sure all data columns are `ListArrays`.
+fn make_all_data_columns_list_arrays(batch: &ArrowRecordBatch) -> ArrowRecordBatch {
+    re_tracing::profile_function!();
 
-    /// The version of the Rerun schema.
-    const CHUNK_METADATA_VERSION: &'static str = "1";
+    let num_columns = batch.num_columns();
+    let mut fields: Vec<ArrowFieldRef> = Vec::with_capacity(num_columns);
+    let mut columns: Vec<ArrowArrayRef> = Vec::with_capacity(num_columns);
 
-    /// The key used to identify a Rerun [`EntityPath`] in the record batch metadata.
-    const CHUNK_METADATA_KEY_ENTITY_PATH: &'static str = "rerun.entity_path";
-}
+    for (field, data) in itertools::izip!(batch.schema().fields(), batch.columns()) {
+        if data
+            .downcast_array_ref::<arrow::array::ListArray>()
+            .is_some()
```

Comment:
This is fairly ambiguous, we do have *components* (e.g. `Blob`) which are themselves lists, but that doesn't necessarily mean they represent a column.

In general, I don't really like the idea of something implicitly casting things to lists like that (for this and other reasons) -- but I'm lacking context, maybe there are good reasons here.

Reply:
We need to find data columns that should be ListArrays (lists of lists), but aren't. Yes, I think we can do better then this.